test_scripts/Capabilities/UpdateVideoStreamingCapabilities/common.lua

```diff
@@ -0,0 +1,460 @@
+      maxBitrate = 1073741823,
+      supportedFormats = {{
+        protocol = "RTP",
+        codec = "VP9"
+      }},
+      hapticSpatialDataSupported = true,
+      diagonalScreenSize = 1000,
+      pixelPerInch = 500,
+      scale = 5.5
+    },
+    [2] = {
+      preferredResolution = {
+        resolutionWidth = 200,
+        resolutionHeight = 200
+      },
+      maxBitrate = 200,
+      supportedFormats = {{
+        protocol = "WEBM",
+        codec = "H265"
+      }},
+      hapticSpatialDataSupported = false,
+      diagonalScreenSize = 200,
+      pixelPerInch = 200,
+      scale = 3
+    }
+  }
+  return utils.cloneTable(videoStreamingCapabilitiesWithOutAddVSC[pIdx])
+end
+
+function m.getVscFromDefaultCapabilitiesFile()
+  return SDL.HMICap.get().UI.systemCapabilities.videoStreamingCapability
+end
+
+function m.buildVideoStreamingCapabilities(pArraySizeAddVSC)
+  if not pArraySizeAddVSC then pArraySizeAddVSC = 1 end
+  local vSC = m.getVscData()
+  vSC.additionalVideoStreamingCapabilities = {}
+  if pArraySizeAddVSC == 0 then
+    vSC.additionalVideoStreamingCapabilities = m.getVscData(2)
+  else
+    for i = 1, pArraySizeAddVSC do
+      vSC.additionalVideoStreamingCapabilities[i] = m.getVscData(2)
+    end
+  end
+  return vSC
+end
+
+function m.setVideoStreamingCapabilities(pVSC)
+  if not pVSC then pVSC = m.buildVideoStreamingCapabilities() end
+  hmiDefaultCapabilities.UI.GetCapabilities.params.systemCapabilities.videoStreamingCapability = pVSC
+end
+
+function m.getSystemCapability(pSubscribe, pAppId, pResponseParams)
+  if not pAppId then pAppId = 1 end
+  if not pResponseParams then pResponseParams = m.buildVideoStreamingCapabilities() end
+  local requestParams = {
+    systemCapabilityType = "VIDEO_STREAMING",
+    subscribe = pSubscribe
+  }
+  local corId = actions.getMobileSession(pAppId):SendRPC("GetSystemCapability", requestParams)
+  actions.getMobileSession(pAppId):ExpectResponse(corId, { success = true, resultCode = "SUCCESS",
+    systemCapability = {
+      systemCapabilityType = "VIDEO_STREAMING"
+    }
+  })
+  :ValidIf(function(_, data)
+    if not m.isTableEqual(pResponseParams, data.payload.systemCapability.videoStreamingCapability) then
+      return false, "Parameters of the response are incorrect: \nExpected: " .. m.toString(pResponseParams)
+      .. "\nActual: " .. m.toString(data.payload)
+    end
+    return true
+  end)
+end
+
+function m.getSystemCapabilityExtended(pAppId, pVsc)
+  pAppId = pAppId or 1
+  pVsc = pVsc or m.buildVideoStreamingCapabilities()
+  local requestParams = {
+    systemCapabilityType = "VIDEO_STREAMING"
+  }
+  local responseParams = {
+    success = true,
+    resultCode = "SUCCESS",
+    systemCapability = {
+      systemCapabilityType = "VIDEO_STREAMING",
+      videoStreamingCapability = pVsc
+    }
+  }
+  local corId = m.getMobileSession(pAppId):SendRPC("GetSystemCapability", requestParams)
+  m.getHMIConnection():ExpectRequest("UI.GetCapabilities"):Times(0)
```

Comment:
Why do we need `m.getHMIConnection():ExpectRequest("UI.GetCapabilities"):Times(0)` here?

Reply:
Here we want to make sure that SDL does not request `UI.GetCapabilities` to get video streaming capabilities while processing of GetSystemCapability request.
`getSystemCapability` function checks `GetSystemCapability` with subscribe parameter.
`getSystemCapabilityExtended` checks `GetSystemCapability` without subscribe parameter and absence of `UI.GetCapabilities`